File: vol2/ch17/LeetCode69.cpp
class Solution {
public:
    int mySqrt(int x) {
        if (x==0) return 0;
        if (x<4) return 1;
        int low=2,high=x/2;
        while(low<=high) {
            int mid=(low+high)/2;
            if (x/mid<mid)  high=mid-1;
            else if (x/(mid+1)<(mid+1))  return mid;
            else low=mid+1;
        }
        return 0;
    }
};

File: vol2/ch17/LeetCode69.py
class Solution:
    def mySqrt(self, x: int) -> int:
        if x==0:return 0
        if x<4:return 1
        low,high=2,x/2
        while low<=high:
            mid=(low+high)//2
            if x/mid<mid:high=mid-1
            elif x/(mid+1)<(mid+1): return int(mid)
            else:low=mid+1
        return 0 
